### PR TITLE
Refactor base station session and screen management

### LINKED ISSUE #32: Base station rendering incorrectly on some platforms
On Surface, the base station appears to use the wrong resolution in full screen mode. The GUI elements are unusable. After minimizing/restoring the window, the window rescales and the issue is resolved. Opening the base station in windowed mode results in an immediate segmentation fault.

### PR DESCRIPTION
**Pull request by superdan-t**
## Summary
Closes #32. The base station no longer manages GLFW itself but relies on NanoGUI to initialize and manage screens. This has greatly reduces the complexity involved in setting up a screen. This refactoring added some new features, like multi-screen support and basic keyboard event handling.
## New features
* There is no longer a main screen associated with the session. An unlimited number of `BasestationScreen` objects can be created.
* Screens process some basic keyboard events
  * Open console: `` ` `` (Grave Accent / Backtick)
  * Toggle fullscreen: `F11`
  * New window: `Ctrl + N`
* Window size settings are now stored in a struct to ease config file creation in the future
* The `src/basestation` is now on the include path for the base station. When including other files, use the full path (eg `#include <controls/controller_manager>`) instead of relative paths (`#include "../controls/controller_manager"`)
## Other findings
* Basically any nanogui window styling change we want to make can be done in nanogui themes. This includes:
  * Disabling window transparency
  * Disabling window borders
  * Using perfectly square borders

### CODE REVIEW

**Review by superdan-t — COMMENTED** [review]
* All changes appear appropriate and necessary to complete this task
* All task requirements have been met
